README.md

```diff
@@ -153,46 +151,41 @@ into the element. This is the only method that must be implemented by subclasses
-  with keys for the changed properties pointing to their previous values.
-  Note, calling `super.update()` is required. Before calling `super.update()`,
-  changes made to properties do not trigger `invalidate()`, after calling `super.update()`,
-  changes do trigger `invalidate()`.
-
-  * `firstRendered()`: (protected) Called after the element's DOM has been
+  * `update(changedProperties)` (protected): This method calls `render()` and then uses `lit-html`
+  in order to render the template DOM. It also updates any reflected attributes based on
+  property values. Setting properties inside this method will *not* trigger another update..
+
+  * `firstUpdated()`: (protected) Called after the element's DOM has been
   updated the first time. This method can be useful for capturing references to rendered static
-  nodes that must be directly acted upon, for example in `update()`.
+  nodes that must be directly acted upon, for example in `updated()`. Setting properties
+  inside this method will trigger the element to update.
+
+  * `updated(changedProperties)`: (protected) Called whenever the element's DOM has been
+  updated and rendered. Implement to perform post updating tasks via DOM APIs, for example,
+  focusing an element. Setting properties inside this method will trigger the element to update.
 
-  * `updateComplete`:  Returns a Promise that resolves when the element has completed
+  * `updateComplete`: Returns a Promise that resolves when the element has completed
   updating that resolves to a boolean value that is `true` if the element completed the
-  update without triggering another update. This can happen if a property is set in
-  `update()` after the call to `super.update()` for example. This getter can be
-  implemented to await additional state. For example, it is sometimes useful to
-  await a rendered element before fulfilling this promise. To do this, first
-  await `super.updateComplete` then any subsequent state.
-
-  * `invalidate`: Call to request the element to asynchronously update regardless
-  of whether or not any property changes are pending. This should only be called
-  when an element should update based on some state not stored in properties,
-  since setting properties automatically calls `invalidate`.
-
-  * `invalidateProperty(name, oldValue)` (protected): Triggers an invalidation for
-  a specific property. This is useful when manually implementing a property setter.
-  Call `invalidateProperty` instead of `invalidate` to ensure that any configured
-  property options are honored.
+  update without triggering another update. This happens if a property is set in
+  `updated()`. This getter can be implemented to await additional state.
+  For example, it is sometimes useful to await a rendered element before fulfilling
+  this promise. To do this, first await `super.updateComplete` then any subsequent state.
+
+  * `requestUpdate(name?, oldValue?)`: Call to request the element to asynchronously
```

Comment:
That method seems to have a third argument `options?`

Reply:
This argument should not be set by users. That's confusing so removing it.